core/services/offchainreporting/delegate.go

```diff
@@ -162,13 +162,14 @@ func (d Delegate) ServicesForSpec(jobSpec job.SpecDB) (services []job.Service, e
 		contractTransmitter := NewOCRContractTransmitter(concreteSpec.ContractAddress.Address(), contractCaller, contractABI,
 			NewTransmitter(d.db.DB(), ta.Address(), d.config.EthGasLimitDefault()))
 
-		ds, err := newDatasource(d.db, jobSpec.ID, d.pipelineRunner)
-		if err != nil {
-			return nil, errors.Wrap(err, "error instantiating data source")
-		}
 		oracle, err := ocr.NewOracle(ocr.OracleArgs{
-			Database:                     NewDB(d.db.DB(), concreteSpec.ID),
-			Datasource:                   ds,
+			Database: NewDB(d.db.DB(), concreteSpec.ID),
+			Datasource: dataSource{
+				pipelineRunner: d.pipelineRunner,
+				jobID:          jobSpec.ID,
+				ocrLogger:      *loggerWith,
+				spec:           *jobSpec.PipelineSpec,
```

Comment:
Worth logging the whole spec? Would the ID be better?

Reply:
I think the request for jobID + contract address as the additional fields was driven by users so I think we just follow that